app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -163,6 +163,14 @@ class BrowserTabViewModel(
         val canFindInPage: Boolean = false
     )
 
+    data class PrivacyGradeViewState(
```

Comment:
I'm revisiting something we already talked about. When `privacyGrade` can be null?

I would like to understand it better, so here are some questions:
- how can I test page cleared? From the test I see that's the scenario where we expect that to be `null`.
- will it be equivalent instead of having attribute `privacyGrade` as `null` to emit `PrivacyGradeViewState` as `null`?

Reply:
This goes back to `WebViewNavigationState` which takes a `WebBackForwardList`. The url is nullable because `WebHistoryItem` is nullable (return null when the list is empty).

- not 100% sure because in theory when the `WebBackForwardList` is empty we are at the home screen but I guess there are probably some edge cases where this could happen. Also if `site` is null for whatever reason, `onSiteChanged` will return `improvedGrade` as null.
- it would complicate things more, we initialize `PrivacyGradeViewState` with the other states as we would normally do, changing that to be initialized to null means having to check whenever we want to use a real state if it was previously null. I think it adds unnecessary complexity.

One option would be to change `onSiteChanged` to return UNKNOWN rather than null but UNKNOWN is an actual grade and so I think returning null is correct because there is no site when that happens.

Reply:
It's ok to keep it like this if we don't know for sure how to reproduce the page cleared. Basically it's hard to find a correct approach if we don't fully understand what's the flow or different possible states.

No need to address this now. We can avoid another refactor 👍